### ISSUE TITLE
tf.12 to tf2.0

### ISSUE THREAD

**Issue by xuchengggg** [tensorflow/tensorflow#34413]
This template is for miscellaneous issues not covered by the other issue categories.

For questions on how to work with TensorFlow, or support for problems that are not verified bugs in TensorFlow, please go to [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow).

If you are reporting a vulnerability, please use the [dedicated reporting process](https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md).

For high-level discussions about TensorFlow, please post to discuss@tensorflow.org, for questions about the development or internal workings of TensorFlow, or if you would like to know how to contribute to TensorFlow, please post to developers@tensorflow.org.

```
def compile(self, learning_rate, momentum):
    """Gets the model ready for training. Adds losses, regularization, and
    metrics. Then calls the Keras compile() function.
    """
    # Optimizer object
    optimizer = keras.optimizers.SGD(
        lr=learning_rate, momentum=momentum,
        clipnorm=self.config.GRADIENT_CLIP_NORM)
    # Add Losses
    # First, clear previously set losses to avoid duplication
    self.keras_model._losses = []
    self.keras_model._per_input_losses = {}
    loss_names = [
        "rpn_class_loss",  "rpn_bbox_loss",
        "mrcnn_class_loss", "mrcnn_bbox_loss", "mrcnn_mask_loss"]
    for name in loss_names:
        layer = self.keras_model.get_layer(name)
        if layer.output in self.keras_model.losses:
            continue
        loss = (
            tf.reduce_mean(layer.output, keepdims=True)
            * self.config.LOSS_WEIGHTS.get(name, 1.))
        self.keras_model.add_loss(loss)

    # Add L2 Regularization
    # Skip gamma and beta weights of batch normalization layers.
    reg_losses = [
        keras.regularizers.l2(self.config.WEIGHT_DECAY)(w) / tf.cast(tf.size(w), tf.float32)
        for w in self.keras_model.trainable_weights
        if 'gamma' not in w.name and 'beta' not in w.name]
    self.keras_model.add_loss(tf.add_n(reg_losses))

    # Compile
    self.keras_model.compile(
        optimizer=optimizer,
        loss=[None] * len(self.keras_model.outputs))

    # Add metrics for losses
    for name in loss_names:
        if name in self.keras_model.metrics_names:
            continue
        layer = self.keras_model.get_layer(name)
        self.keras_model.metrics_names.append(name)
        loss = (
            tf.reduce_mean(layer.output, keepdims=True)
            * self.config.LOSS_WEIGHTS.get(name, 1.))
        self.keras_model.metrics_tensors.append(loss)
```
This code works in tensorflow1.12 with tf.keras, while it does not work in tensorflow 2.0 with tf.keras, where should i change ?  When i turn off eager mode by using `tf.compat.v1.disable_eager_execution()`, it can start to train while there is only loss displayed (without "rpn_class_loss",  "rpn_bbox_loss", "mrcnn_class_loss", "mrcnn_bbox_loss", "mrcnn_mask_loss").

 In eager mode, it raises error using a tf.Tensor as a Python bool is not allowed in Graph execution in `        if layer.output in self.keras_model.losses:
            continue` ; x and y must have the same dtype, got tf.float64!=tf.float32 in `    reg_losses = [
        keras.regularizers.l2(self.config.WEIGHT_DECAY)(w) / tf.cast(tf.size(w), tf.float32)
        for w in self.keras_model.trainable_weights
        if 'gamma' not in w.name and 'beta' not in w.name]`, If i delete those lines, it would have error No gradients provided for any variable.

**Comment by ravikyram** [comment]
Looks like code is incomplete. Request you to provide minimal stand alone code to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by kiflowb777** [comment]
@xuchengggg if you use tf2.0 you have to change: 
`model.metrics_tensors.append(loss)`
to
`model.metrics.append(loss)`

More info: https://github.com/tensorflow/tensorflow/issues/33789

**Comment by ravikyram** [comment]
@xuchengggg 

Can you please confirm if @kiflowb777 's workaround is working for you.Thanks!

**Comment by xuchengggg** [comment]
I am very sorry that the reply is late. I have changed the metrics_tensors to metrics before his @kiflowb777  workaround, but forgot to change the code in the issue and there are still same problems as i mentioned in the issue.
I saw in some tutorials that the source of conda is often updated more slowly and I use `conda install tensorflow-gpu==2.0`, so will it be a problem with the version of conda installed?
